src/StudioComponents/Loading.lua

```diff
@@ -0,0 +1,129 @@
+-- Written by @boatbomber
+
+local Plugin = script:FindFirstAncestorWhichIsA("Plugin")
+local Fusion = require(Plugin:FindFirstChild("Fusion", true))
+
+local StudioComponents = script.Parent
+local StudioComponentsUtil = StudioComponents:FindFirstChild("Util")
+
+local getState = require(StudioComponentsUtil.getState)
+local themeProvider = require(StudioComponentsUtil.themeProvider)
+local constants = require(StudioComponentsUtil.constants)
+local unwrap = require(StudioComponentsUtil.unwrap)
+local types = require(StudioComponentsUtil.types)
+local stripProps = require(StudioComponentsUtil.stripProps)
+
+
+local Computed = Fusion.Computed
+local Hydrate = Fusion.Hydrate
+local Value = Fusion.Value
+local New = Fusion.New
+local Spring = Fusion.Spring
+local Observer = Fusion.Observer
+local Children = Fusion.Children
+
+local COMPONENT_ONLY_PROPERTIES = {
+	"Enabled",
+}
+
+type LoadingProperties = {
+	Enabled: (boolean | types.StateObject<boolean>)?,
+	[any]: any,
+}
+
+local cos = math.cos
+local clock = os.clock
+local pi4 = 12.566370614359172 --4*pi
+
+return function(props: LoadingProperties): Frame
+	local time = Value(0)
+
+	local function startMotion()
+		local startTime = clock()
+		while unwrap(props.Enabled) do
```

Comment:
So this loop stops when `props.Enabled` is set to `false`, but will it stop when the component is destroyed or no longer referenced? Wouldn't it be better to keep track of the last loop coroutine and use `task.close` when it needs to stop and only create a new coroutine if one doesn't exist? And that way you can call `task.close` on the couroutine with `Cleanup` for this component. I have no clue if that makes sense.

Reply:
Just checked, I was wrong. Apparently `task.cancel` works is useable now alongside `task.defer`, `task.spawn`, and `task.defer`.

Reply:
Whoops, forgot to cleanup. Good catch!